utils/clientId_test.go

```diff
@@ -7,21 +7,79 @@ import (
 
 func TestGetIPAddressFromRequest(t *testing.T) {
 
-	expectedResult := "203.0.113.195"
+	expectedIpv4Result := "203.0.113.195"
+	expectedIpv6Result := "2001:9e8:5221:4201:5221:7dff:fe53"
 
 	request := httptest.NewRequest("GET", "/", nil)
+
+	//Test ipv4 address without X-FORWARDED-FOR header (With Port set up)
 	request.RemoteAddr = "203.0.113.195:41237"
-	//First Test without X-FORWARDED-FOR header
 	result := GetIPAddressFromRequest(request)
-	if result != expectedResult {
-		t.Errorf("Remote address only test failed: Expected %s, got %s", expectedResult, result)
+	if result != expectedIpv4Result {
+		t.Errorf("IPV4 Remote address only (with Port) test failed: Expected %s, got %s", expectedIpv4Result, result)
+	}
+
+	//Test ipv4 without X-FORWARDED-FOR header (Without Port set up)
+	request.RemoteAddr = "203.0.113.195"
+	result = GetIPAddressFromRequest(request)
+	if result != expectedIpv4Result {
+		t.Errorf("IPV4 Remote address only (without Port) test failed: Expected %s, got %s", expectedIpv4Result, result)
 	}
 
-	//Test with X-FORWARDED-FOR header
+	//Test ipv4 with X-FORWARDED-FOR header (With Port SetUp)
 	request.Header.Set("X-FORWARDED-FOR header", "203.0.113.195:41237, 198.51.100.100:38523, 140.248.67.176:12345")
```

Comment:
I didn't notice this before so I didn't ask, but why is the word `header` as a part of the `X-FORWARDED-FOR` header?

Reply:
Thank you for noticing this. It was resulting in some tests giving false positives too! It should be resolved now.